e2e/pages/RegistrationDetails/RegistrationDetailsPage.ts

```diff
@@ -238,6 +238,54 @@ class RegistrationDetails {
     ).toContain(userName);
     expect(await timeStampLocator.textContent()).toContain(date);
   }
+
+  async addNote(actions: string, addNote: string) {
+    await this.page.getByText(actions).click();
+    await this.page.getByText(addNote).click();
+  }
+
+  async writeNote(placeholder: string, note: string, buttonName: string) {
+    const okButton = this.page.getByRole('button', {
+      name: buttonName,
+    });
+    await this.page.getByPlaceholder(placeholder).fill(note);
+    await this.page.waitForLoadState('networkidle');
+    await expect(okButton).toBeEnabled();
+
+    for (let i = 0; i < 2; i++) {
+      await okButton.waitFor({ state: 'visible' });
+      await okButton.click();
+    }
```

Comment:
```suggestion
    // accept note
    await okButton.waitFor({ state: 'visible' });
    await okButton.click();
    // dismiss "saved successfully" popup
    await okButton.waitFor({ state: 'visible' });
    await okButton.click();
```

Reply:
Yes this is how I also thought about it. It is a good solution. It was a pickle to decide here but it is more readable now and not so repetitive also it is almost the same amount of line like the loop.